src/threshold-ecdsa/threshold-ecdsa-construction/actual-construction/keygen.md

```diff
@@ -0,0 +1,17 @@
+### Key Generation
+
+In this section, we describe the key generation process in the construction of Gennaro et al. Before moving to the protocol, we provide several notations that will be used in the protocol description below.  
+
+**Notation:** Let \\(\mathbb{G}\\) to be a cyclic group whose order is a prime number. Let \\(p\\) to be the order of \\(\mathbb{G}\\) and let \\(g,h\\) to be two generators of \\(\mathbb{G}\\). We denote \\(\mathsf{Com}\\) to be a secure binding and information theoretic hiding commitment scheme and \\(\mathsf{H}\\) to be a cryptographic hash function. For any set \\(\mathcal{S}\\) and for any \\(i \in \mathcal{S}\\) we denote \\(\lambda_{i,S}=\prod_{j\in \mathcal{S},j \neq i}\dfrac{j}{j-i}\\) to be the Lagrange coefficient w.r.t \\(S\\).
+
+Now, the key generation process is follow.
+
+**Keygen \\((1^\lambda)\langle \\{P_i\\}_{i=1}^n\rangle\\):**
+
+1. Each participant \\(P_i\\) select \\(s_i \in Z_p \\) and compute \\(C_i=\mathsf{Com}(g^{s_i})\\) and broadcasts \\(E_i=(N_i,h_{i1},h_{i2})\\), the public key of Pallier's cryptosystem, where \\(N_i>p^8\\). The Pallier's cryptosystem will be used in the \\(\mathsf{MtA}\\) protocol in [Supporting Protocols](./supporting-algorithms.md).
```

Comment:
`\\(s_i \in Z_p \\)` or `\\(s_i \in Z_q \\)`?

Reply:
s_i in Z_p. In my notation, I define p to be the group order.